File: 3371-identify-the-largest-outlier-in-an-array/3371-identify-the-largest-outlier-in-an-array.cpp
class Solution {
public:
    int getLargestOutlier(vector<int>& nums) {
        int totalSum = accumulate(nums.begin(), nums.end(), 0);
        unordered_map<int, int> mp;
        for (int &x : nums) mp[x]++;        
        int ans = INT_MIN;
        for (int &x : nums) {
            mp[x]--;
            if (mp[x] == 0) {
                mp.erase(x);
            }            
            int rem = totalSum - x;
            if ((rem % 2 == 0) && mp.count(rem / 2)) {
                ans = max(ans, x);
            }
            mp[x]++;
        }        
        return ans;
    }
};

File: LeetCode/3371-identify-the-largest-outlier-in-an-array/3371-identify-the-largest-outlier-in-an-array.py
class Solution:
    def getLargestOutlier(self, nums: List[int]) -> int:
        totalSum = 0
        mp = {}
        for x in nums:
            totalSum += x
            mp[x] = mp.get(x, 0) + 1
        
        ans = float('-inf')
        for x in nums:
            mp[x] -= 1
            if mp[x] == 0:
                del mp[x]
            rem = totalSum - x
            if rem % 2 == 0 and rem // 2 in mp:
                ans = max(ans, x)
            mp[x] = mp.get(x, 0) + 1
        
        return ans